src/commands/diff.rs

```diff
@@ -49,48 +53,76 @@ pub async fn diff(
     // Abort right here if the local Git repository is not clean
     git.check_no_uncommitted_changes()?;
 
+    let mut result = Ok(());
+
     // Look up the commits on the local branch
     let mut prepared_commits = git.get_prepared_commits(config)?;
 
-    let mut prepared_commit = match prepared_commits.pop() {
-        Some(c) => c,
-        None => {
-            output("👋", "Branch is empty - nothing to do. Good bye!")?;
-            return Ok(());
-        }
-    };
-
     // The parent of the first commit in the list is the commit on master that
     // the local branch is based on
-    let master_base_oid = prepared_commits
-        .get(0)
-        .unwrap_or(&prepared_commit)
-        .parent_oid;
+    let master_base_oid = if let Some(first_commit) = prepared_commits.get(0) {
+        first_commit.parent_oid
+    } else {
+        output("👋", "Branch is empty - nothing to do. Good bye!")?;
+        return result;
+    };
 
-    drop(prepared_commits);
+    if !opts.all {
+        // Remove all prepared commits from the vector but the last. So, if
+        // `--all` is not given, we only operate on the HEAD commit.
+        prepared_commits.drain(0..prepared_commits.len() - 1);
+    }
 
-    write_commit_title(&prepared_commit)?;
+    // Fetch Pull Request information from GitHub for all commits in parallel
+    {
+        let futures: Vec<_> = prepared_commits
+            .iter()
+            .filter_map(|prepared_commit| prepared_commit.pull_request_number)
+            .map(|number| gh.get_pull_request(number))
+            .collect();
+        for future in futures {
+            let _ = future.await?;
```

Comment:
would `join!(futures).await?` work here?

Reply:
Ooh maybe... ooph, what crate was `join!` in again?

Reply:
I found it in `futures`, but you can't pass a vector to it.